### ISSUE TITLE
GLIBC_2.28 not found

### ISSUE THREAD

**Issue by XenonSup** [Canop/broot#531]
OS: **Ubuntu 18.04.6 LTS**
Version: **1.11.0**

I've had `broot` installed through https://packages.azlux.fr/ for a while now with no problems.
I updated yesterday from version 1.9.4 to 1.11.0, and now whenever I run `broot` I get the following error message:

```broot: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by broot)```

Running `ldd --version` returns
```ldd (Ubuntu GLIBC 2.27-3ubuntu1.5) 2.27```
So the error doesn't seem erroneous.

I'm under the impression that `glibc` doesn't really get updated that often if at all for a distribution, and that I'm not supposed to try to update it.
Does this mean `broot` doesn't support Ubuntu 18.04 anymore? Or did I break something else?

**Comment by Canop** [comment]
> Does this mean broot doesn't support Ubuntu 18.04 anymore? Or did I break something else?

No, it means I have to find why broot has such requirement and fix it.

**Comment by Canop** [comment]
In the meantime, can you please try the "linux" and "linux/musl" versions that you can download from  https://dystroy.org/broot/install/ ?

**Comment by Canop** [comment]
The underlying problem is probably you use a broot which was compiled for a newer version of glibc. I'm looking for a way to compile for an old glibc on a recent install but it doesn't look easy.

An alternative would be to compile yourself with `cargo install broot` but the rust toolchain takes a lot of space on disk.

**Comment by XenonSup** [comment]
Thank you for the quick response.

> In the meantime, can you please try the "linux" and "linux/musl" versions that you can download from https://dystroy.org/broot/install/ ?

The "Linux" binary gives the same error, but the "Linux/musl" binary seems to work just fine.
This solves my problem of how to use broot, but I'm probably not the only one on an older system so it might still be worth it to look into the dependency change.
Either way, feel free to close the issue.

**Comment by Canop** [comment]
The musl binary should be a little slower, so this isn't a perfect solution.

I'll still need to look for a solution to build for an older glibc, maybe using cargo cross, so I keep the issue open.

**Comment by XenonSup** [comment]
I haven't used the rust development environment so I'll probably wait for the weekend to give it a shot and update then.

**Comment by kabeza** [comment]
Hi
Had the same error installing from packages. Using Elementary Hera 5.1.7 (Ubuntu 18.04 based) and cannot upgrade GLIBC
Removed it and installed manually 1.9.4

**Comment by Canop** [comment]
@XenonSup  @kabeza  Could you please test the version at https://dystroy.org/broot/download/tests/ and confirm it works on your env ?

**Comment by XenonSup** [comment]
Sorry for the late update:

- Installing straight from Crates.io `cargo install broot` results in a working binary (ver 1.11.1)
- Installing from source using the Rust toolchain results in a working binary (ver 1.11.1)
- 1.12.0-dev from https://dystroy.org/broot/download/tests/ works as well

**Comment by Canop** [comment]
Thanks for confirmation. I'm trying to setup a build chain with an old glibc and the right features (current problem is having the clipboard working with the old glibc ^^).

**Comment by kabeza** [comment]
> @XenonSup @kabeza Could you please test the version at https://dystroy.org/broot/download/tests/ and confirm it works on your env ?
> 
> (warning: for tests only)

Hi. Worked fine I guess. I removed old, downloaded and installed manually and did a clean --install

![imagen_0403](https://user-images.githubusercontent.com/274856/163228134-72283a85-8dcb-42c2-9c05-c70c8b8b0f46.jpg)

**Comment by Canop** [comment]
Reopening: will close after official 1.12 release

**Comment by yurenchen000** [comment]
I'm running `broot 1.23.0` prebuild binary (downloaded from github release page).
on `ubuntu 18.04`.

meet same situation. 
// it's fine on ubuntu20, ubuntu22.

<br>

## temporary workaround

as a temporary workaround, 
I have to copy some depends libs (from ubuntu 22) to ubuntu 18, it just work:

```bash
LD_LIBRARY_PATH=./libc.u22 ./libc.u22/ld-linux-x86-64.so.2 ./broot
```

<br>
<br>

I copied those .so
```sh
libc.u22/
├── ld-linux-x86-64.so.2
├── libc.so.6
├── libdl.so.2
├── libm.so.6
├── libpthread.so.0
└── librt.so.1
```

**Comment by Canop** [comment]
Fixed since 1.12: the release archive contains a `x86_64-unknown-linux-gnu.2.28` binary built against an old glibc, and a statically linked `x86_64-unknown-linux-musl` one, for old distributions. Closing.
